docs/rfcs/0020-rotate-hash-peppers.md

```diff
@@ -0,0 +1,379 @@
+   - used HICN lookups sporadically (when MBI failed); mainly limited to synthetic bene(s); 
+## Exploration of Solutions
+[Exploration of Solutions]: #exploration-of-solutions
+
+This section explores five potential strategies for dealing with a required change to BFD's hashing algorithm.
+### Status Quo
+[Status Quo]: #status-quo
+
+Currently, if BFD needs to change the hashing algorithm, it may require some sort of maintenance window and/or minor performance degradation. The current data and the services logic is not capable of handling more than a single instance of the hashing algorithm and a single tuple of a hashed MBI and hashed HICN within each patient record (_BENEFICIARIES_ table). However, there are some simple mechanisms that could be used to at least minimize any downtime.
+
+Since we have no existing _Runbook_ and have no significant work specific to replacing the hash algorithm, the following (rough) pseudo-code steps might be executed in an emergency situation:
+
+- define a new hash algorithm; most likely modify salt (seed value) and/or iteration count (or both). The algorithm construct will need to be communicated to all peering partners in a secure message exchange prior to plan execution and subsequent synchronization of a _go-live_ date at which time BFD would only support newly updated hash values.
+
+- create a new instance (a copy) of the _BENEFICIARIES_ table; say _BENEFICIARIES_PREHASH_; for example, in a flyway script (or possibly out-of-band):
+```
+CREATE BENEFICIARIES_PREHASH AS
+  SELECT * FROM BENEFICIARIES;
+```
+- create and execute a script (python? SQL?) that would:
+  - sequentialy read/update each record in the new table
+  - for each record, calculate new hash value for _MBI_HASH_ and _BENE_CRNT_HIC_NUM_ by hashing respective cleartext columns:
+     - hash _MBI_NUM_ to create _MBI_HASH_ value
+     - hash _HICN_UNHASHED_ to create _BENE_CRNT_HIC_NUM_ 
+  - update (persist) the record
+  - create index for _MBI_HASH_; TBD if we even want/need to create index on _BENE_CRNT_HIC_NUM_.
+```
+CREATE INDEX IF NOT EXISTS beneficiaries_prehash_mbi_idx
+    ON BENEFICIARIES_PREHASH
+    (mbi_hash);
+
+// If we need support for hashed HICN
+CREATE INDEX IF NOT EXISTS beneficiaries_prehash_hicn_idx
+    ON BENEFICIARIES_PREHASH
+    (bene_crnt_hic_num);
+```
+- alert peering partners of scheduled (date-time) maintenance window, at which time the new hash values will be enabled in each environment:
+  - Perform database table rename operations:
+```
+alter table BENEFICIARIES rename to BENEFICIARIES_ORIG;
+alter table BENEFICIARIES_PREHASH rename to BENEFICIARIES;
+```
+  - Perform database indeces rename operations:
+```
+ALTER INDEX beneficiaries_mbi_idx
+  RENAME TO beneficiaries_orig_mbi_idx;
+
+ALTER INDEX beneficiaries_hicn_idx
+  RENAME TO beneficiaries_orig_hicn_idx;
+
+// rename indeces we originally created in the _prehash_ table
```

Comment:
`indeces` is a typo.  Should be `indices` or `indexes`.

Reply:
fixed.